managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerFactoryImpl.java

```diff
@@ -353,6 +353,17 @@ public void asyncOpen(final String name, final ManagedLedgerConfig config, final
                         log.warn("[{}] Attempted to open ledger in {} state. Removing from the map to recreate it",
                                 name, l.getState());
                         ledgers.remove(name, existingFuture);
+                        l.asyncClose(new CloseCallback() {
```

Comment:
Actually, if the managedLedger in `Fenced` or `Closed` state, call `asyncClose` just
https://github.com/apache/pulsar/blob/ee904063abc8ec90c3c536b28a47a59975175cd4/managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerImpl.java#L1316-L1328

So, should we just call `ManagedLedgerFactoryImpl::close` ?

Reply:
Good point. I guess calling asyncClose is sufficient. However there  seems to be a bug in the asyncClose method when the ledger is fenced. The method call to cancel the scheduled tasks is missing.

Reply:
+1